.github/actions/generate-matrix/action.yml

```diff
@@ -10,7 +10,7 @@ inputs:
 runs:
   using: composite
   steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
```

Comment:
question: do we need to checkout again? because the job that calls this already do a checkout.
similar to the other reusable actions

Reply:
Not sure. Might be worth checking in the channel.

Reply:
it is related to using this action in other repos. so yes we need